qlty-cli/src/commands/coverage/complete.rs

```diff
@@ -0,0 +1,403 @@
+            }
+            let project = if let Some(project) = &self.project {
+                project.clone()
+            } else if let Some(repository) = self.find_repository_name_from_env() {
+                repository
+            } else {
+                match self.find_repository_name_from_repository() {
+                    Ok(repository) => repository,
+                    Err(err) => {
+                        debug!("Find repository name: {}", err);
+                        bail!(
+                            "Could not infer project name from environment, please provide it using --project"
+                        )
+                    }
+                }
+            };
+            Ok(format!("{token}/{project}"))
+        } else {
+            Ok(token)
+        }
+    }
+
+    fn find_repository_name_from_env(&self) -> Option<String> {
+        let repository = GitHub::default().repository_name();
+        if repository.is_empty() {
+            None
+        } else {
+            Self::extract_repository_name(&repository)
+        }
+    }
+
+    fn find_repository_name_from_repository(&self) -> Result<String> {
+        let root = Workspace::assert_within_git_directory()?;
+        let repo = Repository::open(root)?;
+        let remote = repo.find_remote("origin")?;
+        if let Some(name) = Self::extract_repository_name(remote.url().unwrap_or_default()) {
+            Ok(name)
+        } else {
+            bail!(
+                "Could not find repository name from git remote: {:?}",
+                remote.url()
+            )
+        }
+    }
+
+    fn extract_repository_name(value: &str) -> Option<String> {
+        value
+            .split('/')
+            .next_back()
+            .map(|s| s.strip_suffix(".git").unwrap_or(s).to_string())
+            .take_if(|v| !v.is_empty())
+    }
+
+    fn load_config() -> QltyConfig {
+        Workspace::new()
+            .and_then(|workspace| workspace.config())
+            .unwrap_or_default()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn complete(project: Option<&str>) -> Complete {
+        Complete {
+            project: project.map(|s| s.to_string()),
+            quiet: true,
+            ..Default::default()
+        }
+    }
+
+    #[test]
+    fn test_expand_token_project() -> Result<()> {
+        let token = complete(None).expand_token("qltcp_123".to_string())?;
+        assert_eq!(token, "qltcp_123");
+        Ok(())
+    }
+
+    #[test]
+    fn test_expand_token_workspace_with_project() -> Result<()> {
+        let token = complete(Some("test")).expand_token("qltcw_123".to_string())?;
+        assert_eq!(token, "qltcw_123/test");
+        Ok(())
+    }
+
+    #[test]
+    fn test_expand_token_workspace_with_env() -> Result<()> {
+        let token = complete(None).expand_token("qltcw_123".to_string())?;
+        assert!(token.starts_with("qltcw_123/"));
+
+        std::env::set_var("GITHUB_REPOSITORY", "");
```

Comment:
Modifying the environment during test execution will create flaky tests. Bad robot!